warehouse/query-core/src/test/java/datawave/query/jexl/functions/EvaluationPhaseFilterFunctionsTest.java

```diff
@@ -476,46 +476,97 @@ private boolean resultForDefaultOperator(int count) {
      * Tests for {@link EvaluationPhaseFilterFunctions#isNotNull(Object)}.
      */
     public static class IsNotNullTests {
-        
-        private Object fieldValue;
+        private Object fieldValue1;
+        private Object fieldValue2;
         
         // Verify that a null object returns an empty set.
         @Test
         public void testNullValue() {
-            givenFieldValue(null);
+            givenFieldValue1(null);
             assertThat(result()).isEmpty();
         }
         
+        // Verify that a null object returns an empty set with multiple filter fields.
+        @Test
+        public void testNullValueWithMultipleFields() {
+            givenFieldValue1(null);
+            givenFieldValue2("FOO.1");
+            
+            assertThat(resultForMultipleFieldValues()).isEmpty();
+        }
+        
         // Verify that a non-null value tuple returns a set with that value tuple.
         @Test
         public void testNonNullValue() {
             ValueTuple valueTuple = toValueTuple("FOO.1,BAR,bar");
-            givenFieldValue(valueTuple);
+            givenFieldValue1(valueTuple);
             assertThat(result()).containsExactly(valueTuple);
         }
         
+        // Verify that a non-null value tuple returns a set with that value tuple with use of multiple filter fields.
+        @Test
+        public void testNonNullValuesWithMultipleFields() {
+            ValueTuple valueTuple = toValueTuple("F00.1,BAR,bar");
+            givenFieldValue1(valueTuple);
+            givenFieldValue2("F00.1");
+            
+            assertThat(resultForMultipleFieldValues()).containsExactly(valueTuple);
```

Comment:
I'm wondering about "F00.1" - should it have been in the resultForMultipleFieldValues?  I'm not very familiar with the use cases for these FilterFunctions so please forgive my ignorance.

Reply:
Is it that value1 and value2's contents overlap?

Reply:
Significant updates to original branch logic.  Please re-review as time permits.  Code in this comment has been replaced.